class Solution {
    public int carFleet(int target, int[] position, int[] speed) {

        int n = position.length;

        // [position, speed]
        int[][] cars = new int[n][2];

        for (int i = 0; i < n; i++) {
            cars[i][0] = position[i];
            cars[i][1] = speed[i];
        }

        // Sort according to position
        Arrays.sort(cars, (a, b) -> Integer.compare(b[0], a[0]));

        int fleetCount = 0;
        double previousTime = 0;

        for (int[] car : cars) {

            double currentTime =
                (double) (target - car[0]) / car[1];

            if (currentTime > previousTime) {
                fleetCount++;
                previousTime = currentTime;
            }
        }

        return fleetCount;
    }
}